src/test/java/testsuite/integration/host/AuroraIntegrationContainerTest.java

```diff
@@ -0,0 +1,187 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package testsuite.integration.host;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.ToxiproxyContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import software.aws.rds.jdbc.mysql.Driver;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AuroraIntegrationContainerTest extends ContainerBaseTest {
+  private static final String DB_CONN_STR_PREFIX = "jdbc:mysql://";
+  private static final String DB_CONN_STR_SUFFIX = System.getenv("DB_CONN_STR_SUFFIX");
+  private static final String DB_CONN_PROP = "?enabledTLSProtocols=TLSv1.2";
+      // Encounters SSL errors without it on GH Actions
+  private static final String TEST_DB_CLUSTER_IDENTIFIER =
+      System.getenv("TEST_DB_CLUSTER_IDENTIFIER");
+  private static final String PROXIED_DOMAIN_NAME_SUFFIX = ".proxied";
+  private static final String RETRIEVE_TOPOLOGY_SQL =
+      "SELECT SERVER_ID FROM information_schema.replica_host_status ";
+  private static final DockerImageName TOXIPROXY_IMAGE =
+      DockerImageName.parse("shopify/toxiproxy:2.1.0");
+  private static final List<ToxiproxyContainer> toxiproxyContainerList = new ArrayList<>();
+  private static final List<String> mySqlInstances = new ArrayList<>();
+
+  private static int mySQLProxyPort;
+  private static GenericContainer<?> integrationTestContainer;
+  private static String dbHostCluster = "";
+  private static String dbHostClusterRo = "";
+
+  private Network network;
+
+  @BeforeEach
+  void setUp() {
+    final String dbConnHostBase =
+        DB_CONN_STR_SUFFIX.startsWith(".")
+            ? DB_CONN_STR_SUFFIX.substring(1)
+            : DB_CONN_STR_SUFFIX;
+    dbHostCluster = TEST_DB_CLUSTER_IDENTIFIER + ".cluster-" + dbConnHostBase;
+    dbHostClusterRo = TEST_DB_CLUSTER_IDENTIFIER + ".cluster-ro-" + dbConnHostBase;
+    network = Network.newNetwork();
+  }
+
+  @AfterEach
+  void tearDown() {
+    for (ToxiproxyContainer proxy : toxiproxyContainerList) {
+      proxy.stop();
+    }
+    integrationTestContainer.stop();
+  }
+
+  @Test
+  public void testRunTestInContainer()
+      throws UnsupportedOperationException, IOException, InterruptedException, SQLException {
+    initializeToxiProxy(network, toxiproxyContainerList);
+    integrationTestContainer = initializeTestContainer(network);
+    runTest(integrationTestContainer, "test-integration-container-aurora");
+  }
+
+  private void initializeToxiProxy(Network network, List<ToxiproxyContainer> toxiproxyContainerList) {
```

Comment:
protected ?
It helps to create other aurora-based test environments by inheriting from this class.